main_test.go

```diff
@@ -102,32 +77,58 @@ func TestSendPreparedTransactionsForPool(t *testing.T) {
 	})
 }
 
-// I leave it here with error: `method eth_syncing` does not exist. I do not want to waste time now for mocking it.
-// Possible solution: add another mock for api for eth and assign method "eth_syncing"
-// and other methods that are missing
-func possibleMockForEthIPC(t *testing.T, ipcLocation string) {
-	myApi := ExternalApiMock{}
+func mockEthIPC(t *testing.T) (path string, close func()) {
+	ipcFile, err := ioutil.TempFile("", "geth_mock_*")
+	if err != nil {
+		log.Fatal(err)
+	}
+	ipcPath := ipcFile.Name()
+
+	myAPI := rpcMethods{}
 
 	rpcAPI := []rpc.API{
 		{
 			Namespace: "account",
 			Public:    true,
-			Service:   myApi,
+			Service:   myAPI,
 			Version:   "1.0",
 		},
 		{
 			Namespace: "eth",
 			Public:    true,
-			Service:   myApi,
+			Service:   myAPI,
 			Version:   "1.0",
 		},
 	}
 
-	listener, server, err := rpc.StartIPCEndpoint(ipcLocation, rpcAPI)
+	l, s, err := rpc.StartIPCEndpoint(ipcPath, rpcAPI)
 	assert.Nil(t, err)
 
-	defer func() {
-		server.Stop()
-		_ = listener.Close()
-	}()
+	return ipcPath, func() {
```

Comment:
Hmm, why we could not have this anonymous function removed and methods done synchronously after line 105?

Reply:
Fixed in #11